Octokit/Models/Response/BranchProtection.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace Octokit
+{
+    /// <summary>
+    /// Protection details for a <see cref="Branch"/>.
+    /// Note: this is a PREVIEW api: https://developer.github.com/changes/2015-11-11-protected-branches-api/
+    /// </summary>
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class BranchProtection
+    {
+        /// <summary>
+        /// Should this branch be protected or not
+        /// </summary>
+        public bool Enabled { get; set; }
```

Comment:
As this is a response object, this will currently fail our checks for public setters.

For serialization purposes, `protected set` will work fine - but we also define `ctors` (like in `Branch` above) for scenarios where you may need to set it up in a certain way.

You can see these failures by running this from the repo root:
 
> .\build BuildApp
> .\build ConventionTests